spec/features/story_show_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+RSpec.describe "story status", js: true do
+  let(:user) { FactoryBot.create(:user) }
+  let(:project) { FactoryBot.create(:project) }
+  let!(:story) { FactoryBot.create(:story, project: project) }
+
+  before do
+    login_as(user, scope: :user)
+  end
+
+  context "from the show page" do
+    it "allows me to change the story status" do
+      visit project_story_path(project_id: project.id, id: story.id)
+
+      status_button = find(".dropdown-wrapper").find("button")
+
+      status_button.click
+      click_button("Approve")
+
+      status_button = find(".dropdown-wrapper").find("button")
+
+      expect(status_button).to have_text("approved")
+    end
+  end
+
+  context "from the estimation modal" do
+    it "allows me to change the story status" do
+      visit project_path(id: project.id)
+      click_link "Add Estimate"
+
+      status_button = find(".story-title").find(".dropdown-wrapper").find("button")
+
+      status_button.click
+      click_button("Approve")
+
+      status_button = find(".story-title").find(".dropdown-wrapper").find("button")
+
+      expect(status_button).to have_text("approved")
+
+      click_button "Save changes"
```

Comment:
The status got updated in the modal even without pressing the "save changes" but on the `show` page it is still the previous status. This might be a little bit weird but okay going with that.

![Peek 2023-10-09 14-23](https://github.com/fastruby/points/assets/7331511/fe64b9eb-f438-4ca3-ba3d-b4ea1919158f)

Reply:
True. I'll change the spec and fix up the JS